scaleio/scaleio.go

```diff
@@ -82,11 +90,18 @@ func (s *ScaleIO) GetMetricTypes(_ plugin.Config) ([]plugin.Metric, error) {
 
 // CollectMetrics implements the collector interface requirements
 func (s *ScaleIO) CollectMetrics(mts []plugin.Metric) ([]plugin.Metric, error) {
-	if s.token == "" || s.client == nil || s.address == nil {
-		err := s.initConnection(mts[0].Config)
+	var client SIOClient
+	gateway, _ := mts[0].Config.GetString("gateway")
```

Comment:
Can you check here for an error ? Would be better for fail here if there is no gateway defined in the slice.